subsys/net/l2/wifi/wifi_utils.c

```diff
@@ -13,26 +13,202 @@ LOG_MODULE_REGISTER(net_wifi_utils, CONFIG_NET_L2_WIFI_MGMT_LOG_LEVEL);
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <ctype.h>
 
 #include <zephyr/kernel.h>
 #include <zephyr/net/net_core.h>
 #include <zephyr/net/wifi.h>
+#include <zephyr/net/wifi_utils.h>
+
 
 /* Ensure 'strtok_r' is available even with -std=c99. */
 char *strtok_r(char *str, const char *delim, char **saveptr);
 
+static const uint16_t valid_5g_chans[] = {32, 36, 40, 44, 48, 52, 56, 60, 64, 68, 96, 100,
+	104, 108, 112, 116, 120, 124, 128, 132, 136, 140, 144, 149, 153, 157, 159, 161,
+	163, 165, 167, 169, 171, 173, 175, 177};
+
 static enum wifi_frequency_bands wifi_utils_map_band_str_to_idx(char *band_str)
 {
+	enum wifi_frequency_bands band = WIFI_FREQ_BAND_UNKNOWN;
+
 	if (!strcmp(band_str, "2")) {
-		return WIFI_FREQ_BAND_2_4_GHZ;
+		band = WIFI_FREQ_BAND_2_4_GHZ;
 	} else if (!strcmp(band_str, "5")) {
-		return WIFI_FREQ_BAND_5_GHZ;
+		band = WIFI_FREQ_BAND_5_GHZ;
 	} else if (!strcmp(band_str, "6")) {
-		return WIFI_FREQ_BAND_6_GHZ;
-	} else {
-		NET_ERR("Unknown band value: %s", band_str);
-		return WIFI_FREQ_BAND_UNKNOWN;
+		band = WIFI_FREQ_BAND_6_GHZ;
+	}
+
+	return band;
+}
+
+
+static bool wifi_utils_validate_chan_2g(uint16_t chan)
+{
+	if ((chan >= 1) && (chan <= 14)) {
```

Comment:
what about regulatory? Should we catch and raise error or as mentioned in the doc, should we let driver/chip handle this? IMHO catching earlier might be better esp. as we have access to `reg_domain`.

Or we can remove all validations here and let driver/chip handle.

Reply:
This PR relies on the driver/chip to handle regulatory validations. However due to the discrete and non-contiguous channel values in 5 GHz we need to impose some checks, especially to derive channel ranges.